lib/streamlit/app_session.py

```diff
@@ -244,13 +244,25 @@ def request_rerun(self, client_state: Optional[ClientState]) -> None:
         else:
             rerun_data = RerunData()
 
-        if self._scriptrunner is None or not self._scriptrunner.request_rerun(
-            rerun_data
-        ):
-            # If we are here, then either we have no ScriptRunner, or our
-            # current ScriptRunner is shutting down and cannot handle a rerun
-            # request - so we'll create and start a new ScriptRunner.
-            self._create_scriptrunner(rerun_data)
+        if self._scriptrunner is not None:
+            if bool(config.get_option("runner.fastReruns")):
+                # If fastReruns is enabled, we don't send rerun requests to our
+                # existing ScriptRunner. Instead, we tell it to shut down. We'll
+                # then spin up a new ScriptRunner, below, to handle the rerun
+                # immediately.
+                self._scriptrunner.request_stop()
+                self._scriptrunner = None
+            else:
+                # fastReruns is not enabled. Send our ScriptRunner a rerun
+                # request. If the request is successful, we're done.
```

Comment:
It would be clearer to describe this as the request being _accepted_, to make it clear that we aren't waiting on the script runner successfully doing the whole script run